packages/media-query/src/use-media-query.ts

```diff
@@ -37,17 +40,18 @@ export function useMediaQuery(
     }))
   })
 
-  useSafeLayoutEffect(() => {
+  useEffect(() => {
     // set initial matches
-    if (ssr) {
-      setValue(
-        queries.map((query) => ({
-          media: query,
-          matches: env.window.matchMedia(query).matches,
-        })),
-      )
-    }
+    setValue(
+      queries.map((query) => ({
+        media: query,
+        matches: env.window.matchMedia(query).matches,
+      })),
+    )
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
```

Comment:
```suggestion
  }, [env.window])
```

We probably want to include the `env.window` in the deps